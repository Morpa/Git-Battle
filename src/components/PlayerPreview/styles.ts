import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 4rem;
    max-width: 50rem;
    color: ${theme.colors.white};
  `}
`
export const Avatar = styled.img`
  ${({ theme }) => css`
    width: 15rem;
    border-radius: 50%;
    margin-top: ${theme.spacings.xsmall};
    margin-bottom: ${theme.spacings.xsmall};
  `}
`
