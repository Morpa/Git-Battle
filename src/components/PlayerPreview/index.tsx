import * as S from './styles'

export type PlayerPreviewProps = {
  avatar: string
  username: string
  label?: string
}

const PlayerPreview = ({ avatar, username, label }: PlayerPreviewProps) => (
  <S.Wrapper>
    <h1>{label}</h1>
    <S.Avatar src={avatar} alt={`Avatar for ${username}`} />
    <h2>{`@${username}`}</h2>
  </S.Wrapper>
)

export default PlayerPreview
