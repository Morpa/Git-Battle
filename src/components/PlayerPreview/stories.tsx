import { Story, Meta } from '@storybook/react/types-6-0'
import PlayerPreview, { PlayerPreviewProps } from '.'

export default {
  title: 'PlayerPreview',
  component: PlayerPreview,
  args: {
    avatar:
      'https://avatars2.githubusercontent.com/u/15898709?s=460&u=86fabb155a6cbb5dc2be016ce890a81e583fdd10&v=4',
    username: 'Morpa'
  }
} as Meta

export const Default: Story<PlayerPreviewProps> = (args) => (
  <PlayerPreview {...args} />
)
