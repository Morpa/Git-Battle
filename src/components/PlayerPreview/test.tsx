import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import PlayerPreview from '.'

const props = {
  avatar:
    'https://avatars2.githubusercontent.com/u/15898709?s=460&u=86fabb155a6cbb5dc2be016ce890a81e583fdd10&v=4',
  username: 'Morpa'
}

describe('<PlayerPreview />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<PlayerPreview {...props} />)

    expect(screen.getByRole('heading', { name: /morpa/i })).toBeInTheDocument()

    expect(
      screen.getByRole('img', { name: `Avatar for ${props.username}` })
    ).toHaveAttribute('src', props.avatar)

    expect(container.firstChild).toMatchSnapshot()
  })
})
