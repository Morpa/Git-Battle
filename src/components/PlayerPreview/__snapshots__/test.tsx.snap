// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerPreview /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4rem;
  max-width: 50rem;
  color: #FFF;
}

.c1 {
  width: 15rem;
  border-radius: 50%;
  margin-top: 1.6rem;
  margin-bottom: 1.6rem;
}

<div
  class="c0"
>
  <h1>
    Player One
  </h1>
  <img
    alt="Avatar for Morpa"
    class="c1"
    src="https://avatars2.githubusercontent.com/u/15898709?s=460&u=86fabb155a6cbb5dc2be016ce890a81e583fdd10&v=4"
  />
  <h2>
    @Morpa
  </h2>
</div>
`
